packages/react-core/src/components/TimePicker/TimePickerUtils.tsx

```diff
@@ -111,3 +110,34 @@ export const getMinutes = (time: string, timeRegex: RegExp) => {
   const parts = time.match(timeRegex);
   return parts && parts.length ? parseInt(parts[2]) : null;
 };
+
+export const isWithinMinMax = (minTime: string, maxTime: string, time: string, delimiter: string) => {
+  // do not throw error if empty string
+  if (time.trim() === '') {
+    return true;
+  }
+  const convertTo24Hour = (time: string): string => {
+    const timeReg = new RegExp(`^\\s*(\\d\\d?)${delimiter}([0-5]\\d)\\s*([AaPp][Mm])?\\s*$`);
+    const regMatches = timeReg.exec(time);
+    if (!regMatches || !regMatches.length) {
+      return;
+    }
+    let hours = regMatches[1].padStart(2, '0');
+    const minutes = regMatches[2];
+    const suffix = regMatches[3] || '';
+    if (suffix.toUpperCase() === 'PM' && hours !== '12') {
+      hours = `${parseInt(hours) + 12}`;
+    } else if (suffix.toUpperCase() === 'AM' && hours === '12') {
+      hours = '00';
+    }
+    return `${hours}${delimiter}${minutes}`;
+  };
+
+  // correctly format as 24hr times (12:30AM => 00:30, 1:15 => 01:15)
+  minTime = convertTo24Hour(minTime);
+  time = convertTo24Hour(time);
+  maxTime = convertTo24Hour(maxTime);
+
+  // simple string comparison for 24hr times
+  return minTime <= time && time <= maxTime;
```

Comment:
In your example, it looks like the max time is not included in the list. Should it be?

Reply:
@nicolethoen it shouldn't be included as it's currently designed, but we could change that.  I'm not touching the time options that are created from the `stepMinutes` other than to remove any that are earlier than the min or greater than the max.

In the example, the `stepMinutes` defaults to 30 minute intervals and because the `maxTime` is 17:15 the last time option shown is 17:00 (as the next option, 17:30, is above the maxTime).

@mcarrano would you expect that we should add an additional time option with their max time (or min time) if it doesn't align with one of the existing time options?  In my example, where `maxTime=17:15`, rather than the last option being `17:00` there would be one more that would be `17:15`?